lib/ts/controllers/s-expandable-control.ts

```diff
@@ -34,92 +32,109 @@
+            return !(<HTMLInputElement>this.element).checked;
+        };
+
 
         get controlledCollapsible() {
-            return document.getElementById(this.element.getAttribute("aria-controls"));
-        },
+            const attr = this.element.getAttribute("aria-controls");
+            if (!attr) {
+                throw "couldn't find controls"
+            }
+            const result = document.getElementById(attr);
+            if (!result) {
+                throw "couldn't find controls"
+            }
+            return result;
+        };
 
-        _dispatchShowHideEvent: function(isShow) {
+        _dispatchShowHideEvent(isShow: boolean) {
             this.triggerEvent(isShow ? "show" : "hide");
-        },
+        };
 
-        _toggleClass: function(doAdd) {
+        _toggleClass(doAdd: boolean) {
             if (!this.data.has("toggle-class")) {
                 return;
             }
             var cl = this.element.classList;
-            this.data.get("toggle-class").split(/\s+/).forEach(function (cls) {
+            var toggleClass = this.data.get("toggle-class");
+            if (!toggleClass) {
+                throw "couldn't find toggle class"
+            }
+            toggleClass.split(/\s+/).forEach(function (cls) {
                 cl.toggle(cls, !!doAdd);
             });
-        },
+        };
 
-        listener: function(e) {
+        listener(e: UIEvent) {
             var newCollapsed;
             if (this.isCheckable) {
-                newCollapsed = !this.element.checked;
+                newCollapsed = !(<HTMLInputElement>this.element).checked;
             } else {
-                if (e.type == "keydown" && (e.keyCode != 13 && e.keyCode != 32)) {
+                if (e.type == "keydown" && ((<KeyboardEvent>e).keyCode != 13 && (<KeyboardEvent>e).keyCode != 32)) {
                     return;
                 }
-                if (e.target !== e.currentTarget && ["A", "BUTTON"].indexOf(e.target.nodeName) >= 0) {
+                if (e.target !== e.currentTarget && ["A", "BUTTON"].indexOf((<HTMLElement>e.target).nodeName) >= 0) {
                     return;
                 }
                 newCollapsed = this.element.getAttribute("aria-expanded") === "true";
                 e.preventDefault();
                 if (e.type === "click") {
-                    this.element.blur();
+                    (<HTMLInputElement>this.element).blur();
                 }
             }
             this.element.setAttribute("aria-expanded", newCollapsed ? "false" : "true");
             this.controlledCollapsible.classList.toggle("is-expanded", !newCollapsed);
             this._dispatchShowHideEvent(!newCollapsed);
             this._toggleClass(!newCollapsed);
-        },
+        };
 
-        connect: function () {
-            this.events.forEach(function (e) {
-                this.element.addEventListener(e, this.listener);
+        connect() {
+            this.events.forEach(e => {
```

Comment:
Just use a cast here, won’t it always be an `EventListener`?

Reply:
as above

Reply:
Addressed as above